### PR TITLE
installing pytorch sometimes fails with OOM on circleCI 

### LINKED ISSUE #1687: CI - Deprecation warnings from pip install plugin/*
example failure https://app.circleci.com/pipelines/github/facebookresearch/hydra/10616/workflows/da305157-4ef7-473e-94f9-abaef33f5d6b/jobs/96090

running the following suggested command raises no warning
```
 pip install --use-feature=in-tree-build plugins/hydra_nevergrad_sweeper/
```

### PR DESCRIPTION
**Pull request by jieru-hu**
Closes #1687

an attempt...

### CODE REVIEW

**Comment by jieru-hu** [comment]
OK - with 8db661d it seems like the ax-sweeper failure goes away, we have one win ray launcher failure but that was due to circleCI infra failure.

**Comment by jieru-hu** [comment]
OK updating the cmmand didnt help, failure coming from 3bf59f6 https://app.circleci.com/pipelines/github/facebookresearch/hydra/10643/workflows/51fdb1f7-b79f-4c02-af5d-ee2e7d4eb1a1/jobs/96354

**Comment by jieru-hu** [comment]
Reproduced in an circleCI instance - but cannot repro this consistently.
```
(hydra) circleci@d3ff18b70e9e:~/project$ pip install plugins/hydra_ax_sweeper/
Processing ./plugins/hydra_ax_sweeper
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting ax-platform>=0.1.20
  Using cached ax_platform-0.1.20-py3-none-any.whl (704 kB)
Collecting torch
Killed
```

**Comment by omry** [comment]
Anything in dmesg on that host? any oomkiller activity?

**Comment by jieru-hu** [comment]
I didn't find any OOM related messages in the /var/log (including dmesg)

I tried to add a step at the end of job to check for memory usage in e64111d and got this [failure](https://app.circleci.com/pipelines/github/facebookresearch/hydra/10651/workflows/9138ec66-19cd-465a-8d7a-89061e5eca48/jobs/96432) 

The memory usage is 4292198400 bytes, which is > 4Gb. Right now our docker image is default which has 4GB memory (https://circleci.com/docs/2.0/configuration-reference/#resourceclass), one thing we could try is to increase the resouce class of the docker image to `medium+` one which has 6GB memory.

**Comment by jieru-hu** [comment]
increase resource class to medium gave us 6GB memory, here is a list of jobs
https://app.circleci.com/pipelines/github/facebookresearch/hydra/10654/workflows/65dd3fc0-ed07-490f-bd54-b8964ee0fb22
version | bytes | GiB
linux-3.9 |6415388672
linux-3.8  6413905920
linux-3.7 6416064512
linux-3.6  2350194688
 
| version      | bytes | GiB
| ----------- | ----------- | ----------- |
| linux-3.9      | 6415388672  | 5.97
| linux-3.8   | 6413905920        |5.97|
| linux-3.7   | 6416064512        |5.98|
| linux-3.6   | 2350194688        |2.19|

File: .circleci/config.yml [hunk 1]
```diff
@@ -202,6 +203,10 @@ jobs:
             export PLUGINS=<< parameters.test_plugin >>
             pip install nox dataclasses --progress-bar off
             nox -s lint_plugins test_plugins -ts
+      - run:
+          name: Check memory usage
+          command: cat /sys/fs/cgroup/memory/memory.max_usage_in_bytes
+          when: always
```
Comment (omry) [inline .circleci/config.yml:209]: Remove this?

File: noxfile.py [hunk 2]
```diff
@@ -386,7 +386,10 @@ def test_plugins_in_directory(
     install_hydra(session, install_cmd)
     selected_plugin = select_plugins(session=session, directory=directory)
     for plugin in selected_plugin:
-        cmd = list(install_cmd) + [os.path.join(directory, plugin.path)]
+        cmd = list(install_cmd) + [
+            os.path.join(directory, plugin.path),
+            "--use-feature=in-tree-build",  # https://github.com/pypa/pip/issues/7555
+        ]
```
Comment (omry) [inline noxfile.py:392]: remove. This will become the default at some point and the error has nothing to do with it.

**Review by omry — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (jieru-hu) [reply .circleci/config.yml]: removed.

File: noxfile.py (hunk 2 above)
Comment (jieru-hu) [reply noxfile.py]: removed.

File: .circleci/config.yml [hunk 3]
```diff
@@ -191,6 +191,7 @@ jobs:
         type: string
     docker:
       - image: cimg/base:stable-18.04
+    resource_class: medium+
```
Comment (omry) [inline .circleci/config.yml:194]: Please add a comment explaining this (or with a link to an issue explaining it).

**Review by omry — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (jieru-hu) [reply .circleci/config.yml:194]: will do :)

**Pull request merged**